src/python/WMCore/Services/WorkQueue/WorkQueue.py

```diff
@@ -325,6 +326,41 @@ def deleteWQElementsByWorkflow(self, workflowNames):
                 deleted += len(ids)
         return deleted
 
+    def getWQElementsByWorkflow(self, workflowNames, inboxFlag=False):
+        """
+        Get workqueue elements which belongs to a given workflow name(s)
+        :param workflowNames: The workflow name for which we try to fetch the WQE elemtns for (could be a list of names as well)
+        :param inboxFlag:     A flag to switch quering the inboxDB as well (default: False)
+        :returnt:             A dictionary with pairs of the type:
+                              {dbName1: [List of WQEs...],
+                               dbName2: [List of WQEs...]}
+        """
+        if inboxFlag:
+            couchdb = self.inboxDB
+        else:
+            couchdb = self.db
+
+        dbName = couchdb.info()['db_name']
+
+        if not isinstance(workflowNames, list):
+            workflowNames = [workflowNames]
+
+        options = {}
+        options["stale"] = "ok"
+        options["reduce"] = False
+
+        wqeList = {}
+        wqeList[dbName] = []
+        result = couchdb.loadView("WorkQueue", "elementsByWorkflow", options, workflowNames)
+
+        ids = []
+        for entry in result["rows"]:
+            ids.append(entry["id"])
+
+        for id in ids:
+            wqeList[dbName].append(couchdb.get(uri=f"/workqueue/_design/WorkQueue/_rewrite/element/{id}" ))
```

Comment:
If I understand this line correctly, here you are actually fetching the workqueue elements given a doc id. Is that correct? If so, why are we doing this? Can't we call the `updateElements()` method when we are actually updating the elements properties (of course, outside this method)?

Reply:
Yes, you did understand properly the line. And again, we are not using this method for updates this is a `get*` method... neither `set*` nor `update*`